src/bin/miri.rs

```diff
@@ -33,20 +33,21 @@ impl rustc_driver::Callbacks for MiriCompilerCalls {
         init_late_loggers();
         compiler.session().abort_if_errors();
 
-        compiler.global_ctxt().unwrap().peek_mut().enter(|tcx| {
-            let (entry_def_id, _) = tcx.entry_fn(LOCAL_CRATE).expect("no main function found!");
-            let mut config = self.miri_config.clone();
+        compiler.enter(|queries| {
+            queries.global_ctxt().unwrap().peek_mut().enter(|tcx| {
```

Comment:
This ICEs. I am not entirely sure why. See https://github.com/rust-lang/rust/pull/66791#issuecomment-559923467.